sql/executor.go

```diff
@@ -54,10 +54,13 @@ func (e *Executor) SetNodeID(nodeID proto.NodeID) {
 // Execute the statement(s) in the given request and return a response.
 // On error, the returned integer is an HTTP error code.
 func (e Executor) Execute(args driver.Request) (driver.Response, int, error) {
+	txnTimestamp := time.Now()
 	planMaker := planner{
 		user: args.GetUser(),
 		evalCtx: parser.EvalContext{
-			NodeID: e.nodeID,
+			NodeID:        e.nodeID,
+			StmtTimestamp: txnTimestamp,
```

Comment:
by the way, this suffers a similar problem to the one described below. Concretely, this will produce the wrong output for this case:

```sql
INSERT INTO t VALUES (statement_timestamp()); SELECT SLEEP (1); SELECT statement_timestamp(), * FROM t;
```